lib/publiccloud/azure.pm

```diff
@@ -147,48 +134,18 @@ sub ipa {
     save_tmp_file($credentials_file, $credentials);
     assert_script_run('curl -O ' . autoinst_url . "/files/" . $credentials_file);
 
-    my $cmd = 'ipa --no-color test azure ';
-    $cmd .= '--debug ';
-    $cmd .= '--service-account-file "' . $credentials_file . '" ';
-    $cmd .= '--distro sles ';
-    $cmd .= '--ssh-private-key-file "' . $args{ssh_private_key_file} . '" ';
-    $cmd .= '--region "' . $self->region . '" ';
-    $cmd .= '--results-dir "' . $args{results_dir} . '" ';
-    $cmd .= ($args{cleanup}) ? '--cleanup ' : '--no-cleanup ';
-    $cmd .= '--instance-type "' . $args{instance_type} . '" ';
-    if (exists($args{running_instance_id})) {
-        $cmd .= '--running-instance-id "' . $args{running_instance_id} . '" ';
-    } else {
-        $cmd .= '--image-id "' . $args{image_id} . '" ';
-    }
-    $cmd .= $args{tests};
-
-    my $output = script_output($cmd . ' 2>&1', $args{timeout}, proceed_on_failure => 1);
-    my $ipa = $self->parse_ipa_output($output);
-    die($output) unless (defined($ipa));
-
-    # retrieves username and password for ssh login
-    $ipa->{username} = 'azureuser';
-    $ipa->{ssh_key}  = $args{ssh_private_key_file};
-
-    $self->{running_instances} //= {};
-    if ($args{cleanup}) {
-        delete($self->{running_instances}->{$ipa->{instance_id}});
-    } else {
-        $self->{running_instances}->{$ipa->{instance_id}} = $ipa;
-    }
-
-    return $ipa;
-}
-
-sub cleanup {
-    my ($self) = @_;
-
-    print Dumper($self->{running_instances});
-    for my $i (keys(%{$self->{running_instances}})) {
-        my $instance = $self->{running_instances}->{$i};
-        $self->ipa(cleanup => 1, running_instance_id => $instance->{instance_id});
-    }
+    return $self->run_ipa(
```

Comment:
Maybe just add additional arguments to `%args` and pass it directly. like
``
$args{user} = 'azureuser';
...;
$self->run_ipa(%args);
``
Same on the other `$self->run_ipa()` calls.